iree/base/math_test.cc

```diff
@@ -0,0 +1,227 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/base/math.h"
+
+#include "iree/testing/gtest.h"
+#include "iree/testing/status_matchers.h"
+
+namespace {
+
+//==============================================================================
+// Bitwise rotation (aka circular shifts)
+//==============================================================================
+
+TEST(BitwiseRotationTest, ROTL64) {
+  EXPECT_EQ(0ull, iree_math_rotl_u64(0ull, 0u));
```

Comment:
Could we have a bit more coverage? Maybe some loop, as in some of the other tests below, to cover more cases with less code.

This isn't to be an insufferable reviewer, but I'm guessing that this is a kind of code that various people will contribute fast/crazy/hardware-specific implementations to in the future, so you want to feel able to take their PRs without doing too much testing mentally everytime ;-)

Reply:
More coverage would be great yes - but there was literally _zero_ coverage before and now there is _some_ that I added while touching this code as part of an incremental code health improvement :)